src/lib/reasoners/fun_sat.ml

```diff
@@ -1810,10 +1750,12 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
       assume env [add_guard env fg,dep]
     with
     | IUnsat (d, classes) ->
-      terminated_normally := true;
       Debug.bottom classes;
       raise (Unsat d)
-    | Util.Timeout when switch_to_model_gen env -> do_switch_to_model_gen env
+    | Util.Timeout ->
+      (* don't attempt to compute a model if timeout before
+         calling unsat function *)
+      raise (I_dont_know {env; timeout = Assume})
```

Comment:
This modification makes sense in the native input language. We don't expect a counterexample of a wrong context. But in the SMT-LIB, we should check what's happened.

Reply:
I think this is the same independent of input language. If the comment is correct, for a SMT-LIB problem, if we timeout here it means that we timed out before processing the `(check-sat)` command, which means that we may not have processed all the assertions / definitions. We shouldn't try to produce a model in this case.

Reply:
You are right. I had a doubt.